flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -174,14 +172,11 @@
 	 * @param partitionSpec partition spec of partition to get
 	 * @return the requested partition
 	 *
-	 * @throws TableNotExistException thrown if the table does not exist in the catalog
-	 * @throws TableNotPartitionedException thrown if the table is not partitioned
-	 * @throws PartitionSpecInvalidException thrown if the given partition spec is invalid,
 	 * @throws PartitionNotExistException thrown if the partition is not partitioned
 	 * @throws CatalogException	in case of any runtime exception
 	 */
 	CatalogPartition getPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
-		throws TableNotExistException, TableNotPartitionedException, PartitionSpecInvalidException, PartitionNotExistException, CatalogException;
+		throws PartitionNotExistException, CatalogException;
```

Comment:
I don't quite get the idea why this function will not throw `TableNotExistException`, but `listPartitions` will. These two methods are both some kind of "get some partition from table".

Reply:
In short, partition not existing includes the case of table not existing. getPartition() attempts to get a specific partition for a table. If the table doesn't exist, then of course, the partition doesn't exist.

On the other hand, listPartition() fails (throws an exception) if the table doesn't exist, which is reasonable.

Just like getTable(), which doesn't throw databasenotexistEx when db doesn't exist. The change makes the behavior consistent across board.